src/compile.ts

```diff
@@ -286,12 +306,22 @@ export function compile(
             toExpr(node.right),
           ]);
         } else if (ts.isPrefixUnaryExpression(node)) {
-          if (node.operator !== ts.SyntaxKind.ExclamationToken) {
+          if (
+            ![
+              ts.SyntaxKind.ExclamationToken,
+              ts.SyntaxKind.MinusToken,
+            ].includes(node.operator)
+          ) {
             throw new Error(
               `invalid Unary Operator: ${ts.tokenToString(node.operator)}`
             );
           }
-          return newExpr("UnaryExpr", [toExpr(node.operand)]);
+          return newExpr("UnaryExpr", [
+            ts.factory.createStringLiteral(
+              ts.tokenToString(node.operator) ?? ""
```

Comment:
Hmm, this default `""` seems wrong?

Reply:
Updated to an assertion and error. This should never happen.